### PR TITLE
Added implementation and test of ballInEnemyDefenseArea and ballInFri…

### LINKED ISSUE #295: Implement nearest_friendly Evaluation Function
### Description of the task

Implement the nearest_friendly evaluation function.

### Outline the task in more detail

We would like to implement the nearest_friendly evaluation function in our new codebase. It is currently defined in the old codebase at `ai/hl/stp/evaluation/team.cpp`. With respect to the interface of this function (ie. what it takes and received), there should be similar types available in the new codebase. That said, it is possible that this function takes a type that is larger then it needs (ex. taking a `World` when it just needs a `Team` and a `ball`). Basically, try to use the smallest type you possible can; this will make it easier to test, review, develop, and work with in the future. 

### Acceptance criteria

- [x] Function nearest_friendly is implemented
- [x] Function nearest_friendly is unit tested

### LINKED ISSUE #299: Implement ball_in_friendly_crease Evaluation Function
### Description of the task

Implement the ball_in_friendly_crease evaluation function.

### Outline the task in more detail

We would like to implement the ball_in_friendly_crease evaluation function in our new codebase. It is currently defined in the old codebase at `ai/hl/stp/evaluation/defense.cpp`. With respect to the interface of this function (ie. what it takes and received), there should be similar types available in the new codebase. That said, it is possible that this function takes a type that is larger then it needs (ex. taking a `World` when it just needs a `Team` and a `ball`). Basically, try to use the smallest type you possible can; this will make it easier to test, review, develop, and work with in the future. 

**NOTE:** We should also implement `ball_in_enemy_crease` since the math should basically be exactly the same

### Acceptance criteria

- [x] Function ball_in_friendly_crease is implemented
- [x] Function ball_in_friendly_crease is unit tested

### PR DESCRIPTION
**Pull request by jonathanlew**
…endlyDefenseArea

### Description
Added implementation and test of ballInEnemyDefenseArea and ballInFriendlyDefenseArea

### Testing Done
Unit tests in world.cpp test file

### Resolved Issues
resolves #299 

### Length Justification
n/a

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/thunderbots/software/test/world/world.cpp [hunk 1]
```diff
@@ -51,6 +51,26 @@ TEST_F(WorldTest, construction_with_parameters)
     EXPECT_EQ(ball, world.ball());
     EXPECT_EQ(friendly_team, world.friendlyTeam());
     EXPECT_EQ(enemy_team, world.enemyTeam());
+
+    // Test logic for world calculations
+    EXPECT_EQ(false, world.ballInFriendlyDefenseArea());
+    EXPECT_EQ(false, world.ballInEnemyDefenseArea());
+    Ball ballfriendly = Ball(Point(-4, 0.5), Vector(-0.3, 0), current_time);
```
Comment (garethellis0) [inline src/thunderbots/software/test/world/world.cpp]: Everything at this line and below should probably be in it's own test. Each unit test should test one specific thing. This test is for the constructor, but the tests for `ballInFriendlyDefenseArea` should have their own (probably multiple) tests.

**Review by garethellis0 — CHANGES_REQUESTED** [review]

File: src/thunderbots/software/test/world/world.cpp (hunk 1 above)
Comment (jonathanlew) [reply src/thunderbots/software/test/world/world.cpp]: OK done

Comment (garethellis0) [reply src/thunderbots/software/test/world/world.cpp]: It looks like we're still testing multiple things in one test? Each test should only encompass a single thing, ie:
- `ball_in_friendly_defense_area`
- `ball_in_enemy_defense_area`
- `ball_just_outside_enemy_defense_area`
- etc.

This makes things much easier to debug if issues arise later.

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]
I don't think this should be a member function of `World`.

I think this function is better generalized to `pointInFriendlyDefenseArea(Point p`). This could either be a member of the `Field` class instead. This way we can perform this evaluation for more than just the ball, and don't need to create and entire `World` just to test it.

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Review by MathewMacDougall — APPROVED** [review]
LGTM :+1:

**Pull request merged**